src/Functions/randomString.cpp

```diff
@@ -0,0 +1,64 @@
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunctionImpl.h>
+#include <Common/thread_local_rng.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int TOO_LARGE_STRING_SIZE;
+}
+
+
+/* Generate random string of specified length with fully random bytes(including zero). */
+class FunctionRandomPrintableASCII : public IFunction
```

Comment:
Are you going to reimplement https://clickhouse.tech/docs/en/sql-reference/functions/other-functions/#randomascii ? 
Have you checked https://clickhouse.tech/docs/en/sql-reference/table-functions/generate/#generaterandom ?
Probably it is much better to add settings that improve `generateRandom` if you don't have specific cases, where random string should be generated by separate code.